src/hdmf/common/resources.py

```diff
@@ -186,88 +197,103 @@ def _add_entity(self, **kwargs):
 
-    @docval({'name': 'resource', 'type': str, 'doc': 'the name of the ontology resource'},
-            {'name': 'uri', 'type': str, 'doc': 'uri associated with ontology resource'})
+    @docval({'name': 'resource', 'type': str, 'doc': 'The name of the ontology resource.'},
+            {'name': 'uri', 'type': str, 'doc': 'The URI associated with ontology resource.'})
     def _add_resource(self, **kwargs):
         """
-        Add resource name and uri to ResourceTable that will be referenced by the ResourceTable idx.
+        Add resource name and URI to ResourceTable that will be referenced by the ResourceTable idx.
         """
         resource_name = kwargs['resource']
         uri = kwargs['uri']
         resource = Resource(resource_name, uri, table=self.resources)
         return resource
 
     @docval({'name': 'container', 'type': (str, AbstractContainer),
-             'doc': 'the Container/Data object to add or the object_id for the Container/Data object to add'},
-            {'name': 'field', 'type': str, 'doc': 'the field on the Container to add'})
+             'doc': 'The Container/Data object to add or the object id of the Container/Data object to add.'},
+            {'name': 'relative_path', 'type': str, 'doc': 'The relative path on the Container to add.'},
+            {'name': 'field', 'type': str, 'default': None,
+             'doc': ('The field of the compound data type using an external resource.')})
     def _add_object(self, **kwargs):
         """
-        Add an object that references an external resource
+        Add an object that references an external resource.
         """
-        container, field = popargs('container', 'field', kwargs)
+        container, relative_path, field = popargs('container', 'relative_path', 'field', kwargs)
         if isinstance(container, AbstractContainer):
             container = container.object_id
-        obj = Object(container, field, table=self.objects)
+        obj = Object(container, relative_path, field, table=self.objects)
         return obj
 
-    @docval({'name': 'obj', 'type': (int, Object), 'doc': 'the Object to that uses the Key'},
-            {'name': 'key', 'type': (int, Key), 'doc': 'the Key that the Object uses'})
+    @docval({'name': 'obj', 'type': (int, Object), 'doc': 'The Object to that uses the Key.'},
+            {'name': 'key', 'type': (int, Key), 'doc': 'The Key that the Object uses.'})
     def _add_object_key(self, **kwargs):
         """
-        Specify that an object (i.e. container and field) uses a key to reference
-        an external resource
+        Specify that an object (i.e. container and relative_path) uses a key to reference
+        an external resource.
         """
         obj, key = popargs('obj', 'key', kwargs)
         return ObjectKey(obj, key, table=self.object_keys)
 
-    def _check_object_field(self, container, field):
+    @docval({'name': 'container', 'type': (str, AbstractContainer), 'default': None,
```

Comment:
When would `container` ever be None here? Also you had the default for `relative_path` and `field` as None and then if they were None, you changed them to `''`. I simplified this by setting the default to `''`.